build.gradle

```diff
@@ -76,7 +76,7 @@ configurations {
 
 dependencies {
 
-    def opensearchVersion = "3.0.0-SNAPSHOT"
+    def opensearchVersion = "2.6.0-SNAPSHOT"
```

Comment:
How are we making sure we keep this updated with the latest OpenSearch 2.x releases?  

When 2.6.0 goes out, this would need to be updated to 2.7.0-SNAPSHOT, but there is no mechanism to let us know this.  A breaking change can go in on 2.7 that we would never see.

We should be able to use Gradle's [`ComponentSelectionRules`](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.ComponentSelectionRules.html) with a `2+` version, only accepting versions with `SNAPSHOT` in the name.  That should get us the highest 2.x SNAPSHOT version.

Reply:
Those rules allow us to read from the snapshot locations?

Reply:
I could have a variable for `opensearchVersion` that's just `2.6.0` and another variable for `opensearchSnapshot` that concatenates the version with `-SNAPSHOT`.  Then we just have to change the version whenever a release goes out, similar to bwc tests

Reply:
I just edited the version like this:
```groovy
    def opensearchVersion = "2.+"
```
And got this:
```
➜  opensearch-sdk-java git:(revert_http_client) ✗ gradle dependencies | grep opensearch-core
|    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    \--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    \--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    \--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    \--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
|    |    +--- org.opensearch:opensearch-core:2.6.0-SNAPSHOT (*)
```
The "+" follows a prefix and I believe matches the newest version with that prefix, be it release or snapshot.  You can further filter it, but I think "newest" woroks here.

Reply:
For "1.+" I get:
```
➜  opensearch-sdk-java git:(revert_http_client) ✗ gradle dependencies | grep opensearch-core
|    +--- org.opensearch:opensearch-core:1.4.0-SNAPSHOT
|    |    +--- org.opensearch:opensearch-core:1.4.0-SNAPSHOT
|    |    \--- org.opensearch:opensearch-core:1.4.0-SNAPSHOT
```

Reply:
Ok so we can change `opensearchVersion` to `2.+`, but we would still need another variable for the tests right?

Reply:
From [the docs](https://docs.gradle.org/current/userguide/single_versions.html):
> A _prefix_ version range: e.g. 1.+, 1.3.+
> 
> Only versions exactly matching the portion before the + are included.
>
> The range + on it’s own will include any version.

That includes snapshots (assuming the snapshot repo is included).  Also,
> Versions are ordered based on the following rules:
>
> Each version is split into it’s constituent "parts":
> 
> The characters [. - _ +] are used to separate the different "parts" of a version.

and then

> The equivalent parts of 2 versions are compared using the following rules:
> 
> If both parts are numeric, the highest numeric value is higher: 1.1 < 1.2
> 
> If one part is numeric, it is considered higher than the non-numeric part: 1.a < 1.1
> 
> If both are non-numeric, the parts are compared alphabetically, in a case-sensitive manner: 1.A < 1.B < 1.a < 1.b
> 
> A version with an extra numeric part is considered higher than a version without (even when it’s zero): 1.1 < 1.1.0
> 
> A version with an extra non-numeric part is considered lower than a version without: 1.1.a < 1.1

So what's happening is it's breaking it into 2, 6, 0, SNAPSHOT which is higher than any 2, 5, anything.  When 2.6.0 is released it will be on the top until 2, 7, anything is available.

So just put "2+". :)

(We should probably switch "main" to just "+" ;)

Reply:
I"m not sure the version in the tests matter (since 2.6 isn't even out).  Does the calling code check the version?

Reply:
Yeah it checks with the recent minimum compatibility check

Reply:
I just checked and it works with previous minor versions, since it's the minimum compatibility.  So we won't have to change this ever